File: api/src/main/java/net/asch/bulkit/api/capability/IDiskResourceRenderer.java
package net.asch.bulkit.api.capability;

import net.minecraft.client.gui.GuiGraphics;
import net.minecraft.network.chat.Component;
import org.jetbrains.annotations.NotNull;

public interface IDiskResourceRenderer {
    @NotNull Component getDescription();
    void render(@NotNull GuiGraphics guiGraphics, int x, int y, int size);
}

File: api/src/main/kotlin/net/asch/bulkit/api/capability/IDiskResourceRenderer.kt
package net.asch.bulkit.api.capability

import net.minecraft.client.gui.GuiGraphics
import net.minecraft.network.chat.Component

interface IDiskResourceRenderer {
    val capacity: Long
    val description: Component

    fun render(guiGraphics: GuiGraphics, x: Int, y: Int, size: Int)
}